From 2485e668761278f1e383a2aaa66823cc42999dde Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Fri, 1 Mar 2013 15:11:48 -0600
Subject: [PATCH 03/36] powerpc: Always init AltiVec exception vector

We weren't initializing the AltiVec Unavailable exception vector
when CONFIG_ALTIVEC was off. This meant that errant AltiVec
instructions sent the kernel into random territory.

Signed-off-by: Andy Fleming <afleming@freescale.com>
Signed-off-by: Kumar Gala <kumar.gala@freescale.com>
[Grabbed from the branch, LINUX_IR5.3.0_ALPHA, of
https://git.freescale.com/git-private/cgit.cgi/ppc/dpaa-offload/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/kernel/exceptions-64e.S |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/arch/powerpc/kernel/exceptions-64e.S b/arch/powerpc/kernel/exceptions-64e.S
index b30ea03..94b4d57 100644
--- a/arch/powerpc/kernel/exceptions-64e.S
+++ b/arch/powerpc/kernel/exceptions-64e.S
@@ -1475,12 +1475,8 @@ _GLOBAL(__setup_base_ivors)
 	blr
 
 _GLOBAL(setup_altivec_ivors)
-#ifdef CONFIG_ALTIVEC
-BEGIN_FTR_SECTION
 	SET_IVOR(32, 0x200) /* AltiVec Unavailable */
 	SET_IVOR(33, 0x220) /* AltiVec Assist */
-END_FTR_SECTION_IFSET(CPU_FTR_ALTIVEC)
-#endif
 	blr
 
 _GLOBAL(setup_perfmon_ivor)
-- 
1.7.5.4

